src/models/utils/selectableFilter.ts

```diff
@@ -15,4 +15,21 @@ export interface SelectableFilter extends Filter {
    * The filter's display name.
    */
   displayName?: string
+}
+
+/**
+ * A {@link CombinedFilter} with additional information, such
+ * as an optional display name and whether or not it is selected.
+ *
+ * @public
+ */
+export interface SelectableCombinedFilter extends CombinedFilter {
```

Comment:
I'd like us to explore some alternatives to this. Adding a `SelectableCombinedFilter` gets the job done, but I think we can come up with a better developer experience. Conceptually, as a Developer, I'd find it a bit odd that there would be two separate models here. A `CombinedFilter` and `Filter` are both static filters. As a Dev, I just want a way to represent a static filter that's selectable. I shouldn't have to pick between selection models depending on how the static filter was constructed. 

What if we changed it so that `SelectableFilter` was a generic that accepted one of `Filter | CombinedFilter`? What would be the ramifications of that?